### PR TITLE
Final bits needed for Drawable

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/client/draw/drawable.h [hunk 1]
```diff
@@ -82,6 +84,66 @@ class DrawableLocoInfo : public MemoryPoolObject
...
+            }
+
+            timing[i] = 0;
+        }
+    }
+    virtual ~DrawableIconInfo() override { Reset(); }
+
+private:
+    Anim2D *anims[MAX_ICONS];
+    unsigned int timing[MAX_ICONS];
```
Comment (xezon) [inline src/game/client/draw/drawable.h]: Maybe word `timings` plural too because `anims` is plural.

File: src/game/client/languagefilter.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "languagefilter.h"
+
+#ifndef GAME_DLL
```
Comment (xezon) [inline src/game/client/languagefilter.cpp:17]: May I suggest to flip the #ifdef to match the one in the header?

File: src/game/client/languagefilter.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+LanguageFilter *g_theLanguageFilter;
```
Comment (xezon) [inline src/game/client/languagefilter.cpp:18]: Does this need to be initialized to something?

File: src/game/logic/object/update/aiupdate.h [hunk 4]
```diff
@@ -17,15 +17,169 @@
...
+
+enum GuardTargetType
+{
+    TARGET_UNK
+};
+enum AIFreeToExitType
+{
+    FREE_TO_EXIT_UNK
+};
+enum GuardMode
```
Comment (xezon) [inline src/game/logic/object/update/aiupdate.h:38]: `GuardModeType` maybe to be consistent with the other enum names here?

File: src/game/logic/object/update/stickybombupdate.h [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    virtual NameKeyType Get_Module_Name_Key() const override;
+    virtual void On_Object_Created() override;
+    virtual void CRC_Snapshot(Xfer *xfer) override;
+    virtual void Xfer_Snapshot(Xfer *xfer) override;
+    virtual void Load_Post_Process() override;
+    virtual UpdateSleepTime Update() override;
+
+    unsigned int Get_Die_Frame() { return m_dieFrame; }
+    bool Has_Die_Frame() { return m_dieFrame != 0; }
+    Object *Get_Target_Object() { return g_theGameLogic->Find_Object_By_ID(m_targetID); }
```
Comment (xezon) [inline src/game/logic/object/update/stickybombupdate.h]: All 3 functions can be `const`.

File: src/game/logic/object/update/stealthupdate.h [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    virtual StealthUpdate *Get_Steath() override;
+    virtual void CRC_Snapshot(Xfer *xfer) override;
+    virtual void Xfer_Snapshot(Xfer *xfer) override;
+    virtual void Load_Post_Process() override;
+    virtual UpdateSleepTime Update() override;
+    virtual BitFlags<DISABLED_TYPE_COUNT> Get_Disabled_Types_To_Process() const override;
+
+    const StealthUpdateModuleData *Get_Stealth_Update_Module_Data() const
+    {
+        return (StealthUpdateModuleData *)Module::Get_Module_Data();
```
Comment (xezon) [inline src/game/logic/object/update/stealthupdate.h]: Hmm is this proper? I would have expected it to allocate a `StealthUpdateModuleData` here.

File: src/game/logic/object/update/stealthupdate.h [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    int m_stealthLevel;
+    unsigned int m_blackMarketCheckDelay;
+    int m_enemyDetectionEvaEvent;
+    int m_ownDetectionEvaEvent;
+    bool m_innateStealth;
+    bool m_orderIdleEnemiesToAttackMeUponReveal;
+    bool m_disguisesAsTeam;
+    bool m_useRiderStealth;
+    bool m_grantedBySpecialPower;
+    friend class StealthUpdate;
```
Comment (xezon) [inline src/game/logic/object/update/stealthupdate.h]: Can we move the friend to the very top? Down here by the members it is kind of obscure.

File: src/game/logic/object/locomotor.h [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+    float m_maximumWheelCompression;
+    float m_wheelTurnAngle;
+    float m_wanderWidthFactor;
+    float m_wanderLengthFactor;
+    float m_wanderAboutPointRadius;
+    float m_rudderCorrectionDegree;
+    float m_rudderCorrectionRate;
+    float m_elevatorCorrectionDegree;
+    float m_elevatorCorrectionRate;
+    friend class Locomotor;
```
Comment (xezon) [inline src/game/logic/object/locomotor.h]: Can we move the friend to the very top? Down here by the members it is kind of obscure.

File: src/game/client/draw/drawable.h [hunk 9]
```diff
@@ -82,6 +84,66 @@ class DrawableLocoInfo : public MemoryPoolObject
...
+            timing[i] = 0;
+        }
+    }
+
+    void Kill_Icon(DrawableIconType icon)
+    {
+        if (anims[icon] != nullptr) {
+            anims[icon]->Delete_Instance();
+            anims[icon] = nullptr;
+            timing[icon] = 0;
```
Comment (xezon) [inline src/game/client/draw/drawable.h]: Should this line be outside of the `anims[icon] != nullptr` scope? In next function it is outside.

File: src/game/client/languagefilter.cpp (hunk 3 above)
Comment (tomsons26) [reply src/game/client/languagefilter.cpp:18]: no

File: src/game/client/draw/drawable.h (hunk 1 above)
Comment (jonwil) [reply src/game/client/draw/drawable.h]: fixed.

File: src/game/logic/object/update/aiupdate.h (hunk 4 above)
Comment (jonwil) [reply src/game/logic/object/update/aiupdate.h:38]: GuardMode is the official name.

File: src/game/logic/object/update/stealthupdate.h (hunk 6 above)
Comment (jonwil) [reply src/game/logic/object/update/stealthupdate.h]: No this is correct, its retrieving a StealhUpdateModuleData that was created earlier via StealthUpdate::Friend_New_Module_Data

File: src/game/client/draw/drawable.h (hunk 9 above)
Comment (jonwil) [reply src/game/client/draw/drawable.h]: This is what the original code did.

File: src/game/logic/object/update/stickybombupdate.h (hunk 5 above)
Comment (jonwil) [reply src/game/logic/object/update/stickybombupdate.h]: fixed.

File: src/game/logic/object/update/stealthupdate.h (hunk 6 above)
Comment (xezon) [reply src/game/logic/object/update/stealthupdate.h]: Ok. `static_cast` then?

File: src/game/logic/object/locomotor.h (hunk 8 above)
Comment (jonwil) [reply src/game/logic/object/locomotor.h]: fixed.

File: src/game/logic/object/update/stealthupdate.h (hunk 7 above)
Comment (jonwil) [reply src/game/logic/object/update/stealthupdate.h]: fixed.

File: src/game/logic/object/update/stealthupdate.h (hunk 6 above)
Comment (jonwil) [reply src/game/logic/object/update/stealthupdate.h]: fixed.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**